analyzer/tests/unit/test_result_handler.py

```diff
@@ -0,0 +1,41 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+import unittest
+
+from codechecker_analyzer.analyzers.result_handler_base import ResultHandler
+
+
+class BuildAction:
+    directory = '/tmp'
+    analyzer_type = 'clangsa'
+    original_command = None
+
+
+class ResultHandlerTestNose(unittest.TestCase):
+    """
+    Test some functions of ResultHandler base class.
+    """
+
+    def test_analyzer_action_str(self):
+        """
+        Check -o removal in .plist filename generation.
+        """
+        ba = BuildAction()
+        rh = ResultHandler(ba, '/tmp/workspace')
+        rh.analyzed_source_file = 'main.cpp'
+
+        ba.original_command = 'g++ main.cpp -o output1 -ooutput2'
+        self.assertEqual(
+            rh.analyzer_action_str,
+            'main.cpp_clangsa_b42298618a535959e9adc7807414763c')
```

Comment:
It's hard to tell from the generated file name whether the output argument is hashed or not. A better test case would be if you put a random string (current date or something) to the original build command instead of a static string (output1).

Reply:
I don't understand this request. Why does a random string in build command make generated file name more understandable? This will always be a hash value which is actually comes from "g++ main.cpp". However, it would be useful creating other test-cases where output flags are placed in different order: "g++ -o output1 main.cpp -ooutput2" and the generated hash must be the same. What do you think?

Reply:
The reason why you created this patch because make will generate a random output file name every time you run it. So the build command will be changed every time.
Here comes my problem. Here you test a static string here (`g++ main.cpp -o output1 -ooutput2`) which doesn't contain any random output file. So how do I know that the generated hash (`b42298618a535959e9adc7807414763c`) doesn't contain the `-o output1` string?